File: scripts/migrate-chars.py
import sqlite3
import os

db_path = r"F:\自媒体\night-nest\apps\api\prisma\dev.db"
conn = sqlite3.connect(db_path)
cur = conn.cursor()

migrations = [
    ("imageUrl", "TEXT DEFAULT ''"),
    ("profile", "TEXT DEFAULT '{}'"),
    ("creatorId", "TEXT DEFAULT 'system'"),
]

for col, dtype in migrations:
    try:
        cur.execute(f"ALTER TABLE Character ADD COLUMN {col} {dtype}")
        print(f"Added column: {col}")
    except sqlite3.OperationalError as e:
        print(f"Skip {col}: {e}")

conn.commit()
conn.close()
print("Done.")

File: scripts/migrate-chars.js
const { PrismaClient } = require("@prisma/client");
const p = new PrismaClient({ datasources: { db: { url: "file:F:/自媒体/night-nest/apps/api/prisma/dev.db" } } });

async function main() {
  const migrations = [
    "ALTER TABLE Character ADD COLUMN imageUrl TEXT DEFAULT ''",
    "ALTER TABLE Character ADD COLUMN profile TEXT DEFAULT '{}'",
    "ALTER TABLE Character ADD COLUMN creatorId TEXT DEFAULT 'system'",
  ];
  for (const sql of migrations) {
    try { await p.$executeRawUnsafe(sql); console.log("OK:", sql.substring(0, 50)); }
    catch (e) { console.log("Skip:", e.message.substring(0, 60)); }
  }
  p.$disconnect();
}
main();
